dm/syncer/syncer.go

```diff
@@ -2040,14 +2040,20 @@ func (s *Syncer) handleRowsEvent(ev *replication.RowsEvent, ec eventContext) err
 	if err != nil {
 		return terror.WithScope(err, terror.ScopeDownstream)
 	}
-	rows, err := s.mappingDML(sourceTable, tableInfo, ev.Rows)
+	originRows, err := s.mappingDML(sourceTable, tableInfo, ev.Rows)
 	if err != nil {
 		return err
 	}
 	if err2 := checkLogColumns(ev.SkippedColumns); err2 != nil {
 		return err2
 	}
 
+	extRows := generateExtendColumn(originRows, s.tableRouter, sourceTable, s.cfg.SourceID)
+	rows := originRows
+	if extRows != nil {
+		rows = extRows
+	}
+
 	prunedColumns, prunedRows, err := pruneGeneratedColumnDML(tableInfo, rows)
```

Comment:
Hi, since [ti.Columns](https://github.com/pingcap/ticdc/blob/27424e8bf420ad461268e38cd1edb6a47931319c/dm/syncer/dml.go#L459) contains extend columns, `extRows` and `ti.Columns` are the same length.

Reply:
Ok, at that time I thought the table structure was init from dump file. But I still prefer the tableinfo should be same as upstream. cc @lance6716

Reply:
I left some comments [here](https://github.com/pingcap/ticdc/pull/3262#issuecomment-975290984) about why use the downstream table structure. PTAL